synapse/lib/cell.py

```diff
@@ -1679,6 +1679,8 @@ async def promote(self, graceful=False):
             mesg = 'promote() called on non-mirror'
             raise s_exc.BadConfValu(mesg=mesg)
 
+        logger.warning(f'PROMOTION: Performing leadership promotion {graceful=} ahaname={self.conf.get("aha:name")}')
+
         if graceful:
 
             ahaname = self.conf.get('aha:name')
```

Comment:
Seems like we might as well grab `aha:name` before the first warning to avoid grabbing it from `self.conf` a bunch of times in the log messages

Reply:
Changed this; but left the `ahaname is None` check in the graceful codepath. In a mirror case where aha isn't being used and the leader has gone away; you can still force a promotion that isn't graceful.